test_data/test_postprocessor/source/page4.txt

```diff
@@ -1 +1,7 @@
 .. include:: /includes/test.rst
+
+Using |sub| before definition.
+
+.. |sub| replace:: Substitution
+
+Using |sub| after definition.
```

Comment:
I think we optionally might consider making this test a little more comprehensive:
* Define a substitution in an include, use it here
* Use `|sub|` in an include
* If `|sub|` is redefined elsewhere in the page, what evaluation order is used in determining the value at any given point?

Reply:
Good points here! Regarding point 3, current behavior is that the substitution definition is overwritten whenever the same `|sub|` is encountered. Overwriting is based on the parse order of the page which is... I'm not sure! Ultimately all references to `|sub|` are replaced with the same value. Do you think this is logical behavior? For the record, implementing another way seems potentially _much_ harder.